src/main/java/net/dv8tion/jda/api/utils/SessionController.java

```diff
@@ -144,10 +143,81 @@
      * @return The Pair consisting of the gateway endpoint to connect to and the shardTotal
      *
      * @see    #getGateway(net.dv8tion.jda.api.JDA)
+     *
+     * @deprecated
+     *         Use {@link #getShardedGateway(JDA)} instead, an implementation for this is ignored
+     *         if {@link #getShardedGateway(JDA)} is implemented instead.
      */
     @Nonnull
+    @Deprecated
+    @ForRemoval
+    @DeprecatedSince("4.0.0")
+    @ReplaceWith("getShardedGateway(api)")
+    @SuppressWarnings("DeprecatedIsStillUsed")
     Pair<String, Integer> getGatewayBot(@Nonnull JDA api);
 
+    /**
+     * Called by {@link net.dv8tion.jda.api.sharding.DefaultShardManager DefaultShardManager}
+     * when a new shards is starting.
+     * <br>Should provide a {@link GatewayBot} with {@code (gateway, shardTotal)}.
+     *
+     * @param  api
+     *         The current JDA instance (used for RestActions and ShardInfo)
+     *
+     * @return The GatewayBot instance consisting of the gateway endpoint to connect to and the shardTotal
+     *
+     * @see    #getGateway(net.dv8tion.jda.api.JDA)
+     */
+    @Nonnull
+    default GatewayBot getShardedGateway(@Nonnull JDA api)
+    {
+        Pair<String, Integer> tuple = getGatewayBot(api);
+        return new GatewayBot(tuple.getLeft(), tuple.getRight());
+    }
+
+    /**
+     * POJO containing the gateway endpoint and recommended shard total for a shard manager.
+     */
+    class GatewayBot
```

Comment:
Should this still be called `GatewayBot` since we're moving to the `ShardedGateway` naming?